### PR TITLE
chore: add nucleus config @W-14756916

### PR DESCRIPTION
**Pull request by wjhsf**
Pretty much as advertised. Copied from [LWC](https://github.com/salesforce/lwc/blob/2870fc339e0d710b3d29ece8cb3d3e8923f55e17/.nucleus.yaml), with a few tweaks.

### CODE REVIEW

File: .nucleus.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        # - omnistudio/ui-flexcards-components # Failed on 'install dependencies'
+        # - omnistudio/ui-omniscript-components # Failed on 'install dependencies'
+        # - omnistudio/ui-omnistudio-components # Failed on 'install dependencies'
+        # - omnistudio/ui-omnistudiobase-components # Failed on 'install dependencies'
+        # - salesforce-experience-platform-emu/lightning-components # Long-running
+        # - salesforce-experience-platform-emu/lws-apps # Failed on 'install dependencies'
+        # - Skilling-and-Enablement/ui-external-enablement # Failed on 'run unit test'
+        # - Skilling-and-Enablement/ui-in-app-guidance # Failed on 'run unit tests'
+        # - Skilling-and-Enablement/ui-ses-experience-cloud # Failed on 'run unit test'
+  (winter|spring|summer)[2-9][0-9]:
```
Comment (wjhsf) [inline .nucleus.yaml:41]: Confluence says regex is supported!

File: .nucleus.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        # - communities/ui-b2b-components # Failed on 'Run LCT tests'
+        # - communities/ui-cms-components # Failed on 'build project'
+        # - communities/ui-dxp-components # Failed on 'Run tests against example app'
+        # - communities/ui-feeds-components # Failed on 'build project'
+        # - LightningMobileRuntime/lsdk-modules # Failed on 'install dependencies'
+        # - omnistudio/ui-flexcards-components # Failed on 'install dependencies'
+        # - omnistudio/ui-omniscript-components # Failed on 'install dependencies'
+        # - omnistudio/ui-omnistudio-components # Failed on 'install dependencies'
+        # - omnistudio/ui-omnistudiobase-components # Failed on 'install dependencies'
+        # - salesforce-experience-platform-emu/lightning-components # Long-running
```
Comment (wjhsf) [inline .nucleus.yaml:36]: This has not finished after ~90 minutes, while every other project either succeeded or failed in under 20 minutes (and mostly under 10).

File: .nucleus.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+branches:
+  ~DEFAULT~:
+    pull-request: &branch-definition
+      auto-start: true
+      auto-start-from-forks: false
+      merge-method: disabled # do not auto-merge; we'll do it ourselves
+      validate-optional-downstream-deps: true # temporarily test non-required downstreams
+      required-downstream-deps:
```
Comment (wjhsf) [inline .nucleus.yaml:8]: This is everything that ran for #229. All packages that passed are listed; all that failed are not.

Comment (nolanlawson) [reply .nucleus.yaml:8]: Do we have a sense of the flappiness of these? Was this based on just one test run?

**Review by nolanlawson — APPROVED** [review]
LGTM, although I am a little nervous about flappers.

File: .nucleus.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+branches:
+  ~DEFAULT~:
+    pull-request: &branch-definition
+      auto-start: true
+      auto-start-from-forks: false
+      merge-method: disabled # do not auto-merge; we'll do it ourselves
+      validate-optional-downstream-deps: true # temporarily test non-required downstreams
```
Comment (nolanlawson) [inline .nucleus.yaml:7]: FWIW this became permanent in `salesforce/lwc` – it was too useful to keep it on. :)

File: .nucleus.yaml (hunk 1 above)
Comment (jmsjtu) [reply .nucleus.yaml:41]: Posting for posterity, in the past nucleus has had [issues with using regex ](https://salesforce-internal.slack.com/archives/C022MRLQ9R6/p1670446506983039?thread_ts=1670442242.320659&cid=C022MRLQ9R6) for branch definitions.  I'm not sure if it's still an issue but might be something to look at if anything pops up in the future.

**Review by jmsjtu — APPROVED** [review]

File: .nucleus.yaml (hunk 3 above)
Comment (wjhsf) [reply .nucleus.yaml:8]: Just one test run.

**Pull request merged**